enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,887 @@
+can't bring up any additional infrastructure at the cluster site, in particular
+I can't bring up an image registry server, neither outside of the cluster nor
+inside. I want to plug the USB stick provided by the engineer in the node and
+have the rest of the process performed automatically.
+
+#### Upgrade a multi-node cluster
+
+As a technician with little or no OpenShift knowledge I want to be able to
+upgrade a multi-node cluster as well. This is the same story than for
+single-node clusters but for multi-node clusters. The difference is that I don't
+want to plug the USB stick to all the nodes of the cluster, instead I want to
+plug it to only one of the nodes (selected randomly) and have the rest of the
+process performed automatically, included the propagation of the bundle contents
+to all the nodes.
+
+#### Pre-load upgrade images
+
+As an engineer managing a cluster that has a low bandwidth and/or unreliable
+connection to an image registry server I want to pre-load all the images
+required for the upgrade so that when I decide to actually perform the upgrade
+there will be no need to contact that slow and/or unreliable registry server.
+
+### Goals
+
+Provide an automated and documented mechanism that partner engineers customer
+technicians can use to upgrade a cluster without requiring a registry server.
+
+### Non-Goals
+
+It is not a goal to not require a registry server for other operations. For
+example, installing a new workload will still require a registry server.
+
+## Proposal
+
+### Workflow Description
+
+1. An engineer working in a partner's factory is asked to prepare a bundle to
+upgrade a set of clusters to a specific OpenShift version.
+
+1. The engineer uses the `oc adm upgrade create bundle ...` tool described in
+this enhancement to prepare the upgrade bundle containing all the artifacts
+(container images and metadata) that are required to perform the upgrade, and
+writes it USB stick (or any other suitable media) that will be handed over to
+the technicians responsible for performing the upgrades, together with
+documentation explaining how to use it.
+
+1. The technicians receive copies of the USB stick and the corresponding
+documentation.
+
+1. The technician goes to the cluster site and uses the upgrade bundle inside
+the USB stick to perform the upgrade. The documentation included in the bundle
+will basically ask the technician to plug the USB stick in one of the nodes of
+the cluster and then provide simple instructions to verify that the upgraded has
+been applied correctly. This step is potentially repeated multiple times by the
+same technician for multiple clusters using the same USB stick or copies of it.
+
+Note that the upgrade bundle should not be specific for a particular cluster,
+only for the OpenShift architecture and version. Technicians should be able to
+use that package for any cluster with that architecture and version.
+
```

Comment:
Please also speak about the need to include partner images in that upgrade bundle.

Reply:
This is already mentioned in the "Include custom images in the bundle" user story, and in the section that describes the `oc adm upgrade create bundle` command. I added a reference here.